renesas: rz-cmn: board-specific overrides via DT

tiennguyenuh opened the pull request:
This change makes board bring-up portable across RZ/G2L, RZ/V2L, and RZ/V2H by moving board-specific pokes into DT and applying them in a generic way.

vudangRVC commented on arch/arm/dts/rzg2l-sbc.dts:
Adding a "board-overrides" node unnecessarily complicates board-specific implementations.

All board-specific code should remain in the `s_init` function for early initialization and the `late_init` function for late function calls. This approach ensures consistency, making it easier for users to add new logic for a new board without worrying about where to place it.

tiennguyenuh replied:
Removed.

vudangRVC commented on board/renesas/rz-cmn/rz-cmn.c:
```diff
@@ -220,17 +240,16 @@ int setup_uboot_info_from_qspi(void)
+		case RZ_SOC_RZV2H:
 			ret = spi_flash_read(flash, QSPI_BOARD_INFO_OFFSET_V2H, CONFIG_ENV_SIZE, board_info);
 			break;
-		case BOARD_ID_RZV2L_EVK:
-		case BOARD_ID_RZG2L_EVK:
-		case BOARD_ID_RZG2L_SBC:
+		case RZ_SOC_RZG2L:
+		case RZ_SOC_RZV2L:
 			ret = spi_flash_read(flash, QSPI_BOARD_INFO_OFFSET, CONFIG_ENV_SIZE, board_info);
 			break;
 		default:
-			printf("Runtime: unknown or unsupported board_id = %llu\n", board_id);
+			printf("Runtime: unknown or unsupported soc_id = %llu\n", soc_id);
 			ret = -EINVAL;
 			goto cleanup;
 	}
```
I'm okay with these changes. Making this funtion as SoC-specific implementation is understandable

vudangRVC commented on board/renesas/rz-cmn/rz-cmn.c:
```diff
@@ -294,21 +313,33 @@ int setup_uboot_info_from_qspi(void)
 
 int board_fit_config_name_match(const char *name)
```
This function is essential when adding a new board. Please keep the old code and mention this function in the porting guide

tiennguyenuh replied:
I removed this function because multiple device trese in the FIT image are no longer used.

vudangRVC commented on drivers/mmc/sh_sdhi.c:
Please confirm if these changes are SoC-specific.

tiennguyenuh replied:
Yes, it's SoC-specific.

tiennguyenuh committed:
plat: renesas: remove FIT support and refine SoC-specific conditions

Changes:
	- Remove FIT match configuration, as multiple device trees
	  are not required in a single U-Boot build. Flashing script
	  will handle combining the correct DTB.
	- Convert certain conditions from board-specific to SoC-specific
	  for better maintainability.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
